fix: Fixed links for older releases of Installer

jakub-kocka opened the pull request:
Based on the reported issue, the fix for the older IDF Installer releases in [Espressif's download server](https://dl.espressif.com/dl/esp-idf/) is provided.

* Fixed automatic tag and asset name generation to correspond with the links
* Changed the tag and asset names already released to make the links work

## Related
* Closes https://github.com/espressif/idf-installer/issues/254

dobairoland reviewed (approved):
LGTM with the exception that we will need the mirrored URLs to be fixed in the near future.

jakub-kocka commented:
Thank you @dobairoland for the review, I will leave this PR open, to add the Mirror links fix later as well (it will need a bit of more work).

In case we need to release a new IDF Installer, I will merge this to avoid more bad links to GH and will open another for the fix of the Mirror links.

jakub-kocka commented:
With Tomas's help, we have updated links in the download server and I have updated the "URL generator" to correspond with these links for the `Mirror` button.
